### ISSUE TITLE
Visually inconsistent text markup / font style

### ISSUE THREAD

**Issue by dennismayr** [jrblevin/markdown-mode#358]
# Visually inconsistent text markup / font style

Using the same syntax for different elements on a list, the fonts weight or style is rendered differently.

## Expected Behavior

Visually consistent font weight color / style.

## Actual Behavior

Different font style / color for the elements on a list, using the same syntax for each.

## Steps to Reproduce

Make a list of different elements under the same hierarchical indentation:
- This is element n° 1: `code snippet goes here`
- This is element n° 2: `code snippet goes here`
- This is element n° 3: `code snippet goes here`
- This is element n° 4: `code snippet goes here`

Image:
![Visually consistent font weight color / style](https://user-images.githubusercontent.com/30353238/44357060-809b4600-a487-11e8-87f2-96fb93d27fb7.png)

Not a show-stopper, but still strange to look at.

- Markdown Mode: 20180731.1830
- Emacs: 26.1 (although this has happened since emacs24 and emacs25)
- OS: Linux, Ubuntu 16.04, kernel 4.4.0-133

**Comment by dennismayr** [comment]
Is there anything I can do to help debug this?
Just let me know.

**Comment by jrblevin** [comment]
Yes, sorry, I haven't had much free time to dedicate to markdown-mode in a while. If you are able to narrow it down or find a patch that doesn't break any existing tests then that would definitely help. I would start with `markdown-match-code`. It looks like it is having trouble matching across lines in your example (which is difficult in general for Emacs font-lock). There is a great package called font lock studio that might also help. There are a ton of font lock rules in markdown-mode, so you could disable (comment out) irrelevant ones to make stepping through them faster.

**Comment by dennismayr** [comment]
No worries; I’ll look into it during the week; I’m trying to debug an Emacs problem that involves `markdown-mode` buffers using `minimap`, crashing Emacs+Spacemacs with *segfault*, so this goes on the list as well.

**Comment by syohex** [comment]
![test](https://user-images.githubusercontent.com/554281/83580698-a9ff6600-a577-11ea-909c-e396cebe919c.png)

I cannot reproduce. Does this issue still happens ?

**Comment by john-cj** [comment]
Yes, the issue still exists.

```
* foo1: `key1: value2`
* bar2: `key3: value4`
```

![image](https://user-images.githubusercontent.com/18004961/127931423-c617a04e-5c8f-4382-b861-5aefaccec527.png)

This is especially weird when you type the text inside fenced code block. Not only the font face is changed to very thin font, but the line height is changed as well, multiple times. 

![emacs](https://user-images.githubusercontent.com/18004961/127932094-9c0d1567-ef97-4180-9eb0-0ea04524cfc0.gif)

GNU Emacs 27.1 (build 1, i686-w64-mingw32) of 2020-08-21

**Comment by dennismayr** [comment]
Here's another example:

Check lines 286 and 287, how the commands between `backticks` are correctly highlighted, and line 289 isn't, but lines 290 and 291 are inversely highlighted, line 295 is correctly highlighted again, and line 296 is inversely highlighted.

![Captura de pantalla de 2022-05-28 11-58-04](https://user-images.githubusercontent.com/30353238/170833170-a93f30d6-c262-44f5-b72d-655d5279159f.png)

**Comment by syohex** [comment]
Could you show us raw text instead of the screenshot ? It's hard to type.

**Comment by Martinsos** [comment]
This happens to me for even very simple examples.

It is as if randomly only some parts are colored. 

By doing small modifications, like adding a whitespace and then removing it, I get coloring to refresh and it turns out in some other state this time.

For example, here I managed to get three different colorings for this file (none of them completely correct) by just adding and removing lines to the file (always ending up at the same state). So I would add and remove the empty line, and I get different coloring:

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/99785b31-054a-45a1-98ee-e972e826974c)

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/40f580ad-e800-4ade-8a2e-4b2477b8c64e)

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/19b99236-5d10-450d-b266-d140d72e5307)

Here is the example I used:
```md

 - `:%s/query/new/gc`

 - `g r m`     -> make all cursors
 - `g r q`     -> delete all cursors

 - `g w`       -> evil fill -> needs motion

```

This is happening on Archlinux, emacs 29.1, spacemacs, markdown-mode 2.7-alpha.

```
Minor modes enabled in this buffer: Anzu Auto-Save Company Eldoc Emoji-Cheat-Sheet-Plus-Display
Evil-Local Evil-Mc Evil-Surround Font-Lock Hl-Todo Spacemacs-Leader-Override Subword Undo-Tree
Vi-Tilde-Fringe Yas
```
